### PR TITLE
Consul rule file namess match those for k8s

### PR DESCRIPTION
**Pull request by GregHanson**
**Release note**:

```release-note
<PLEASE PUT RELEASE-NOTE HERE, PUT NONE if NO RELEASE-NOTE>
NONE
```

### CODE REVIEW

**Comment by rshriram** [comment]
While you are at this, move the normal rules into rules/kube/ directory. Similarly, move these into rules/consul/ directory. That would involve some minor changes in test scripts (i.e. changing hardcoded paths to use the correct rules/kube.

cc @frankbu 

In another pr, we could trim the samples/apps/bookinfo to just samples/bookinfo

**Comment by frankbu** [comment]
Why not get rid of the `bookinfo/rules` dir and just put the platform specific stuff (just rules currently) in `bookinfo/kube` and `bookinfo/consul`?

File: samples/bookinfo/consul/docker-compose.yaml [hunk 1]
```diff
@@ -51,7 +51,7 @@ services:
     command: ["-internal", "-retry-attempts=-1", "consul://consul:8500"]
 
   discovery:
-    image: $HUB/discovery:latest
+    image: istio/discovery:latest
```
Comment (rshriram) [inline samples/bookinfo/consul/docker-compose.yaml]: nix

**Comment by GregHanson** [comment]
breaking PR into smaller ones

**Comment by GregHanson** [comment]
see #896

**Pull request closed without merging**